pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -87,20 +97,79 @@ func (c *APIClient) connect() error {
 	return err
 }
 
-func parseKubeConfig(fpath string) (*k8s.Config, error) {
-	yamlFile, err := ioutil.ReadFile(fpath)
+// ParseKubeConfig reads and unmarcshals a kubeconfig file
+// in an object ready to use. Exported for integration testing.
+func ParseKubeConfig(fpath string) (*k8s.Config, error) {
+	// TODO: support yaml too
+	jsonFile, err := ioutil.ReadFile(fpath)
 	if err != nil {
 		return nil, err
 	}
 
 	config := &k8s.Config{}
-	err = yaml.Unmarshal(yamlFile, config)
+	err = json.Unmarshal(jsonFile, config)
 	return config, err
 }
 
+// ComponentStatuses returns the component status list from the APIServer
 func (c *APIClient) ComponentStatuses() (*v1.ComponentStatusList, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 	return c.client.CoreV1().ListComponentStatuses(ctx)
+}
+
+// ConfigMapTokenFetcher returns the value of the `tokenValue` from the `tokenKey` in the ConfigMap `configmaptokendca` if its timestamp is less than tokenTimeout old.
+func (c *APIClient) ConfigMapTokenFetcher(tokenKey string) (string, bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	defer cancel()
+	tokenConfigMap, err := c.client.CoreV1().GetConfigMap(ctx, configMapDCAToken, defaultNamespace)
+	if err != nil {
+		return "", false, err
+	}
+	log.Infof("Found the ConfigMap %s", configMapDCAToken)
+
+	token, found := tokenConfigMap.Data[tokenKey]
+	if !found {
+		return "", found, log.Errorf("%s was not found in the ConfigMap %s", tokenKey, configMapDCAToken)
+	}
+	log.Debugf("%s is %s", tokenKey, token)
+
+	tokenTimeStr, set := tokenConfigMap.Data[fmt.Sprintf("%s.%s", tokenKey, tokenTime)] // This is so we can have one timestamp per tokenKey
+
+	if set {
+		tokenTime, err := strconv.Atoi(tokenTimeStr)
```

Comment:
you should use the `time` package's methods instead of converting on your own. See https://play.golang.org/p/ubuGVolITGb